lib/archethic/networking/port_forwarding.ex

```diff
@@ -70,4 +75,27 @@ defmodule Archethic.Networking.PortForwarding do
   defp ip_lookup_provider do
     Application.get_env(:archethic, IPLookup)
   end
+
+  @iana_registry_path Path.join([
+                        File.cwd() |> elem(1),
+                        "lib",
+                        "archethic",
+                        "networking",
+                        "registry",
+                        "iana_unassigned_ports"
+                      ])
+
+  # see https://github.com/apoorv-2204/iana_port_registry
+  defp list_iana_unassigned_ports() do
```

Comment:
I think you could simplify without the need to list the port from IANA. Between 49152 through 65535 should make it. 
The randomness could be recursive until one port is free.